### DISCUSSION TITLE
Different validation for different steps

### DISCUSSION THREAD

**Discussion by SOUMAJYOTI** [Lightning-AI/pytorch-lightning#17518] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I have a use case where I want to use different validation code in 

> validation_step(...)

  depending on the training step count. For example, at the end of every 100, 300, 500,... training steps, I want to call one validation code whereas for all 200, 400, .. steps I want to use a different validation code. Is there a work around for this?

**Comment by miladfa7** [comment]
Here's an example implementation:
```
def validation_step(self, batch, batch_idx):
    # use different validation code for different steps
    if self.global_step % 100 == 0 or self.global_step % 300 == 0 or self.global_step % 500 == 0:
        # execute validation code for steps 100, 300, 500, ...
        val_loss = ...
    elif self.global_step % 200 == 0 or self.global_step % 400 == 0:
        # execute validation code for steps 200, 400, ...
        val_loss = ...
    else:
        # execute default validation code
        val_loss = ...
    return {'val_loss': val_loss}

```
In this implementation, you can check the current training step using the global_step property of the LightningModule class. If the step is a multiple of 100, 300, or 500, we execute the first validation code, and if it is a multiple of 200 or 400, we execute the second validation code. Otherwise, we execute the default validation code.
```
def validation_step(self, batch, batch_idx):
    # use different validation code for different epochs
    if self.current_epoch == 0:
        # execute validation code for first epoch
        val_loss = ...
    elif self.current_epoch == 1:
        # execute validation code for second epoch
        val_loss = ...
    else:
        # execute default validation code
        val_loss = ...
    return {'val_loss': val_loss}

```
